charts/kyverno/templates/_helpers.tpl

```diff
@@ -116,3 +116,27 @@ maxUnavailable: {{ .Values.podDisruptionBudget.maxUnavailable }}
 {{- define "kyverno.imagePullSecret" }}
 {{- printf "{\"auths\":{\"%s\":{\"auth\":\"%s\"}}}" .registry (printf "%s:%s" .username .password | b64enc) | b64enc }}
 {{- end }}
+
+
+{{- define "kyverno.resourceFilters" -}}
+{{- $resourceFilters := .Values.config.resourceFilters }}
+{{- if .Values.excludeKyvernoNamespace }}
+  {{- $resourceFilters = prepend .Values.config.resourceFilters (printf "[*,%s,*]" (include "kyverno.namespace" .)) }}
+{{- end }}
+{{- tpl (join "" $resourceFilters) . }}
+{{- end }}
+
+{{- define "kyverno.webhooks" -}}
+{{- $excludeDefault := dict "key" "kubernetes.io/metadata.name" "operator" "NotIn" "values" (list (include "kyverno.namespace" .)) }}
+{{- $newWebhook := list }}
+{{- range $webhook := .Values.config.webhooks }}
+  {{- $namespaceSelector := default dict $webhook.namespaceSelector }}
+  {{- $matchExpressions := default list $namespaceSelector.matchExpressions }}
+  {{- $newNamespaceSelector := dict "matchLabels" $namespaceSelector.matchLabels "matchExpressions" (append $matchExpressions $excludeDefault) }}
+  {{- if not (empty (omit $webhook "namespaceSelector")) }}
+    {{- $newWebhook = append $newWebhook (omit $webhook "namespaceSelector") }}
+  {{- end }}
+  {{- $newWebhook = append $newWebhook (dict "namespaceSelector" $newNamespaceSelector)}}
```

Comment:
This still does not look correct. Only the first item in the list will be considered by the controller so having two append statements is probably not going to give us the expected behavior.

Reply:
The first append statement is omitting the namespaceSelector so it would be just be objectSelector.  The second one appends the new namespaceSelector.